src/nix-env/nix-env.cc

```diff
@@ -1336,16 +1336,26 @@ static void opListGenerations(Globals & globals, Strings opFlags, Strings opArgs
     if (opArgs.size() != 0)
         throw UsageError("no arguments expected");
 
+#ifndef _WIN32 // TODO Implement path locks on Windows.
     PathLocks lock;
     lockProfile(lock, globals.profile);
+#endif
 
     auto [gens, curGen] = findGenerations(globals.profile);
 
     RunPager pager;
 
     for (auto & i : gens) {
+#ifdef _WIN32 // TODO portable wrapper in libutil
```

Comment:
This could probably be made portable using `std::chrono`, though the time zone stuff is C++20.

Reply:
I'll make this a good-first-issue follow-up task, since it is localized.